releasenotes-builder/src/test/java/com/github/checkstyle/internal/AbstractReleaseNotesTestSupport.java

```diff
@@ -0,0 +1,201 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.internal;
+
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.contrib.java.lang.system.SystemErrRule;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.kohsuke.github.GHFileNotFoundException;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHIssueState;
+import org.kohsuke.github.GHRepository;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import com.github.checkstyle.Main;
+import com.github.checkstyle.git.CsGit;
+import com.github.checkstyle.github.CsGitHub;
+import com.github.checkstyle.publishers.TwitterPublisher;
+
+public abstract class AbstractReleaseNotesTestSupport extends AbstractPathTestSupport {
+
+    private static final Set<RevCommit> TEST_COMMITS = new LinkedHashSet<>();
+
+    private static final Set<GHIssue> TEST_ISSUES = new LinkedHashSet<>();
+
+    private static final Answer<GHIssue> ISSUE_ANSWER = new Answer<>() {
+        @Override
+        public GHIssue answer(InvocationOnMock invocation) throws Throwable {
+            final int findIssue = invocation.getArgument(0);
+
+            for (GHIssue item : TEST_ISSUES) {
+                if (item.getNumber() == findIssue) {
+                    return item;
+                }
+            }
+
+            throw new GHFileNotFoundException("Cannot find Issue: " + findIssue);
+        }
+    };
+
+    private static boolean mocked;
+
+    @Rule
+    public final SystemErrRule systemErr = new SystemErrRule().enableLog().mute();
+    @Rule
+    public final SystemOutRule systemOut = new SystemOutRule().enableLog().mute();
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        if (!mocked) {
+            mocked = true;
```

Comment:
As found out in other PR, we can only mock the classes once. This prevents double mocking.